Explicitly document ability to implement multiple of the same hook (#2256)
Napari/napari: docs/plugins/for_plugin_developers.rst

   def whatever_name_you_want(path: str):
      ...

This allows you to specify multiple hook implementations of the same hook 

> specification in the same file, without needing an entirely new module or entry point.

Reviewer: ```suggestion
specification in the same module or class, without needing a separate entry point.
```

I think it's good to keep remembering/reminding that a plugin is a namespace, not necessarily just a file.  

As a sidenote: @DragaDoncila, this may be particularly useful for you to know as you investigate this stuff.  For example, to confirm this multi-plugin-registration functionality, you could do something like this:

```python
from napari_plugin_engine import (
    HookImplementationMarker,
    HookSpecificationMarker,
    PluginManager,
)

hookimpl = HookImplementationMarker("test")
hookspec = HookSpecificationMarker("test")

class HookSpecs:
    @hookspec
    def my_hook_spec(path: str):
        ...

class Plugin:
    @hookimpl
    def my_hook_spec(path: str):
        ...

    @hookimpl(specname='my_hook_spec')
    def my_hook_spec_number2(path: str):
        ...

pm = PluginManager("test")
pm.add_hookspecs(HookSpecs)
pm.register(Plugin, name='test_plugin')
assert 'test_plugin' in pm.plugins
for impl in pm.hooks.my_hook_spec.get_hookimpls():
    print(f"Plugin {impl.plugin_name}: {impl.function}")
```

Author: Excellent point, I've removed the references to "in the same file" for the experimental hookspec docstrings as well.
